libraries/AP_Math/vector3.cpp

```diff
@@ -251,7 +251,9 @@ void Vector3<T>::rotate(enum Rotation rotation)
         z = -sin_pitch * tmpx + cos_pitch * tmpz;
         return;
     }
-    case ROTATION_CUSTOM: // no-op; caller should perform custom rotations via matrix multiplication
+    default:
```

Comment:
This shouldn't be a default case, as then the compiler can't catch people adding an enumeration value but not adding a case here for it.

Putting it after the switch statement here will work, 'though - means if the compiler doesn't catch things - for example, the value coming from a parameter and that parameter being out of range - you'd still get the failure.


You'd want a simple "break"  for `ROTATION_CUSTOM` then.

Reply:
Ah, I had forgotten that the compiler checked for that; thought it would just do nothing, which would be bad.

Reply:
Thanks, I moved the internalError call out of the switch as suggested so static checks are unchanged and the only behavior change is to call internalError if the rotation is unsupported.
SITL will now panic if ROTATION_CUSTOM is selected, because it doesn't handle that correctly.